### ISSUE TITLE
Tracking issue for HA to beta

### ISSUE THREAD

**Issue by fabriziopandini** [kubernetes/kubeadm#1567]
What is missing for getting HA to beta in v1.15?

- [x] “--experimental-control-plane” flag to be renamed “--control-plane”
- [x] “--experimental-upload-certs” flag to be renamed “--upload-certs”. Default to true?
- [x] https://github.com/kubernetes/kubeadm/issues/1071 “kubeadm upgrade node experimental-control-plane” to be merged into “kubeadm upgrade node”
- [punted to v1.16] https://github.com/kubernetes/kubeadm/issues/1432 make` join --control-plane` more resilient (it should work after loosing one control plane/etcd) https://github.com/kubernetes/kubernetes/pull/75600
- [x] e2e tests
   - [x] change kind-master job (and possibly kind-1.14) to test HA
   - [x] change upgrade-stable-master job (and possibly  upgrade-1.13-1.4) to test HA + automatic copy certificates
   - [x]  change master-on-stable job (and possibly 1.14 on 1.13) to test HA + automatic copy certificates
   - [x]  create a new job testing HA + automatic copy certificates with external etcd

**Comment by fabriziopandini** [comment]
@neolit123 opinions about the E2E part? the main idea behind is to not recreate a new series of job for HA, but instead, make current jobs have an additional control plane node so they are testing also HA

**Comment by neolit123** [comment]
>  “--experimental-upload-certs” flag to be renamed “--upload-certs”. Default to true?

needs discussion with the group.

> “kubeadm upgrade node experimental-control-plane” to be merged into “kubeadm upgrade node”

i'd be interested in how the detection will work here.

> @neolit123 opinions about the E2E part? the main idea behind is to not recreate a new series of job for HA, but instead, make current jobs have an additional control plane node so they are testing also HA

+1 on switching everything to HA.

> create a new job testing HA + automatic copy certificates with external etcd

seems good. i think we can get away without upgrade for external etcd for the beta...

**Comment by neolit123** [comment]
> +1 on switching everything to HA.

also i can work on these if you'd like.

**Comment by RA489** [comment]
@fabriziopandini i can work on some of them.
 “--experimental-control-plane” flag to be renamed “--control-plane”
  “--experimental-upload-certs” flag to be renamed “--upload-certs”. Default to true?

**Comment by neolit123** [comment]
hi @RA489 we spoke about these during yesterday's kubeadm office hours.
the idea was to do these PR last from the list from the list above.

also i think instead of renaming `--experimental-control-plane` -> `--control-plane` it might be better to have a transitional period of supporting both frags at the same time - e.g. at least 3 months.

**Comment by RA489** [comment]
@neolit123 thanks for the explanation. let me know if i can help with some others PR

**Comment by RA489** [comment]
@fabianofranz  i can look into change kind-master job (and possibly kind-1.14) to test HA

**Comment by RA489** [comment]
@fabriziopandini I can look into change kind-master job (and possibly kind-1.14) to test HA

**Comment by neolit123** [comment]
>  create a new job testing HA + automatic copy certificates with external etcd

this is the only remaining item - PR is in flight waits on test-grid merge approval:
https://github.com/kubernetes/test-infra/pull/12906

> [punted to v1.16] #1432 make join --control-plane more resilient (it should work after loosing one control plane/etcd) kubernetes/kubernetes#75600

this is a nice to have marked as `priority/backlog` and can be a post-beta improvement.

**Comment by neolit123** [comment]
only remaining item is:
> create a new job testing HA + automatic copy certificates with external etcd

https://github.com/kubernetes/test-infra/pull/12906

and this issue can be closed :tada:

**Comment by fabriziopandini** [comment]
/close 
As per comment above the latest open issue be a post-beta improvement and it is already tracked in #1432
